File: perl/unit_testing/test_basic_math.pl
#!/usr/bin/perl

use diagnostics;
use strict;
use warnings;
use Test::More qw( no_plan );

do './basic_math.pl';

is(hello(), 'hello', 'test_hello');
is(min(), 2, 'test_min');
is(max(), 3.4, 'test_max');

File: python/unit_testing/test_basic_math.py
from basic_math import *
import unittest

class TestBasicMath(unittest.TestCase):

    def test_add(self):
        self.assertEqual(add(2,2), 4)

    def test_min(self):
        self.assertEqual(min(2,5),2)

    def test_max(self):
        self.assertEqual(max(8, 4), 8)

if __name__ == '__main__':
    unittest.main()
